### PR TITLE
Fix type reference name in attribute argument getting mangled when

### PR DESCRIPTION
**Pull request by mrvoorhe**
such attribute is put on a method in a projected type.

This is a fix that Unity's cecil has had for a couple years.

### CODE REVIEW

File: Test/Mono.Cecil.Tests/TypeParserTests.cs [hunk 1]
```diff
@@ -88,7 +88,7 @@ public void ByRefTypeReference ()
 		public void FullyQualifiedTypeReference ()
 		{
 			var module = GetCurrentModule ();
-			var cecil = module.AssemblyReferences.Where (reference => reference.Name == "Mono.Cecil").First ();
+			var cecil = module.AssemblyReferences.Where (reference => reference.Name == "Unity.Cecil").First ();
```
Comment (jbevain) [inline Test/Mono.Cecil.Tests/TypeParserTests.cs:91]: Nope :)

Comment (mrvoorhe) [reply Test/Mono.Cecil.Tests/TypeParserTests.cs]: Fixed

**Comment by mrvoorhe** [comment]
I can't tell if a test failed or if there was some CI instability.  All I see in the log is
```
  Determining projects to restore...
D:\a\cecil\cecil\symbols\pdb\Mono.Cecil.Pdb.csproj : error NU1101: Unable to find package Microsoft.SourceLink.GitHub. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\a\cecil\cecil\Mono.Cecil.sln]
D:\a\cecil\cecil\symbols\pdb\Mono.Cecil.Pdb.csproj : error NU1101: Unable to find package Microsoft.NETFramework.ReferenceAssemblies.net40. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\a\cecil\cecil\Mono.Cecil.sln]
  Failed to restore D:\a\cecil\cecil\symbols\pdb\Mono.Cecil.Pdb.csproj (in 1.1 sec).
  Restored D:\a\cecil\cecil\symbols\pdb\Test\Mono.Cecil.Pdb.Tests.csproj (in 49.83 sec).
  Restored D:\a\cecil\cecil\symbols\mdb\Test\Mono.Cecil.Mdb.Tests.csproj (in 48.71 sec).
  Restored D:\a\cecil\cecil\symbols\mdb\Mono.Cecil.Mdb.csproj (in 7 ms).
  Restored D:\a\cecil\cecil\Test\Mono.Cecil.Tests.csproj (in 125 ms).
  Restored D:\a\cecil\cecil\rocks\Mono.Cecil.Rocks.csproj (in 5 ms).
  Restored D:\a\cecil\cecil\rocks\Test\Mono.Cecil.Rocks.Tests.csproj (in 127 ms).
  Restored D:\a\cecil\cecil\Mono.Cecil.csproj (in 5 ms).

Build FAILED.

```

**Comment by mrvoorhe** [comment]
Looks like it was a CI instability.  I force pushed to trigger a new CI run and now the tests are green.

**Comment by jbevain** [comment]
Yeah that's NuGet having hiccups.

File: Mono.Cecil/AssemblyWriter.cs [hunk 2]
```diff
@@ -2998,6 +2998,27 @@ void WriteCustomAttributeValue (TypeReference type, object value)
 			}
 		}
 
+		private void WriteCustomAttributeTypeValue (TypeReference value)
+		{
+			var typeDefinition = value as TypeDefinition;
+			TypeDefinition outermostDeclaringType = typeDefinition;
```
Comment (jbevain) [inline Mono.Cecil/AssemblyWriter.cs:3004]: Could you move this into the if branch?

Comment (mrvoorhe) [reply Mono.Cecil/AssemblyWriter.cs]: Done

**Review by jbevain — APPROVED** [review]

**Comment by jbevain** [comment]
Thank you!

**Pull request merged**